burn-import/src/onnx/op_configuration.rs

```diff
@@ -81,6 +82,41 @@ pub fn conv2d_config(curr: &Node) -> Conv2dConfig {
     .with_padding(padding)
 }
 
+/// Create a MaxPool2dConfig from the attributes of the node
+pub fn max_pool2d_config(curr: &Node) -> MaxPool2dConfig {
+    let mut channels: i64 = 1;
+    let mut kernel_shape = Vec::new();
+    let mut strides = Vec::new();
+    let mut pads = Vec::new();
+
+    for (key, value) in curr.attrs.iter() {
+        match key.as_str() {
+            "channels" => attr_value_i64(value, &mut channels),
+            "kernel_shape" => attr_value_vec_i64(value, &mut kernel_shape),
+            "strides" => attr_value_vec_i64(value, &mut strides),
+            "pads" => attr_value_vec_i64(value, &mut pads),
+            _ => {}
+        }
+    }
+
+    let padding = if pads.iter().all(|&x| x == 0) {
+        MaxPool2dPaddingConfig::Valid
+    } else {
+        todo!("max_pool2d: padding({pads:?}) is not fully supported");
```

Comment:
It'd be great if we could implement padding as well. The same logic could be used by the Conv2d, since they're the same.

Reply:
The main problem with `padding` is that we can have something like this as node: `[2, 2, 2, 2]` and looking at `Conv2dPaddingConfig` I do not know which enum field I should choose for this case. [Here](https://github.com/burn-rs/burn/blob/main/burn-core/src/nn/conv/conv2d.rs#L43) I should choose `Explicit` for this case, but I have only two values and not four. And whens should I choose the `Same` field?

Reply:
I put a `panic!("{:?}", node)` instruction for the `MaxPool` type and I got this:

```
Node { node_type: MaxPool, name: "/model.9/m/MaxPool", inputs: [Argument { name: "/model.9/cv1/act/Mul_output_0", ty: Tensor(TensorArg { dim: 0 }) }], outputs: [Argument { name: "/model.9/m/MaxPool_output_0", ty: Tensor(TensorArg { dim: 0 }) }], states: [], attrs: {"ceil_mode": Int64(0), "pads": Int64s([2, 2, 2, 2]), "strides": Int64s([1, 1]), "kernel_shape": Int64s([5, 5])} }', burn-import/src/onnx/from_onnx.rs:384:13
```

As you can see `"pads": Int64s([2, 2, 2, 2])` has four values

Reply:
> Yeah you can use explicit, but it's strange that you have more than 2 values for a 2d convolution 😅.

Basically they're left, up, right, and bottom (need to verify which is which). Here is the full document: https://onnx.ai/onnx/operators/onnx__Pad.html

Burn and [Pytorch's padding ](https://pytorch.org/docs/stable/generated/torch.nn.MaxPool2d.html) does support the full padding like in ONNX. Most likely all exported ONNX padding can be supported by Burn's current implementation.

Therefore, we should only extract values we are interested for and implement `ToTokens` for MaxPool2dPaddingConfig in `burn-import/src/burn/codegen.rs`. If we encounter, asymmetric top/bottom or left/right, then we will same we do not support it. Basically, we will support examples, such as `[2, 2, 2, 2]` or [2, 2, 3, 3], which will be equivalent as `Conv2dPaddingConfig::Explicit(2,2)` and `Conv2dPaddingConfig::Explicit(2, 3). 
`
@Luni-4, I hope this makes sense. I am standby if you need further help. If you could, tackle this conversion, the same can be used for Conv padding in next PRs easily - win/win =D